### ISSUE TITLE
Redirect JSON file Error

### ISSUE THREAD

**Issue by MertcanGokgoz** [TryGhost/Ghost#9036]
### Issue Summary

While uploading the file I get an error like below, is happening during upload completing the upload gives

no problem with other uploads(images etc)

### Steps to Reproduce

1. Login ghost admin
2. Select Labs
3. Download current redirects file
4. Edit json file
5. Upload valid json file from computer

`Cannot read property 'invalidFile' of null`

### Technical details:

* Ghost Version: 1.9.0
* Node Version: 6.11.3
* Browser/OS: Chrome 61.0.3163.91
* Database: Mysql

redirect.json file

```
[{
		"permanent": true,
		"from": "([^/.]+).html$",
		"to": "/$1"
	},
	{
		"permanent": true,
		"from": "^/category/([a-z0-9\\-]+)?",
		"to": "/tag/$1"
	}
]
```

Screenshot

![error](https://user-images.githubusercontent.com/8042604/30723938-256d6910-9f43-11e7-8b9f-a0c429d3f191.PNG)

**Comment by vikaspotluri123** [comment]
Looks like the actual problem is a fileType error, but there was an issue with translating the error so you get an exception as the error message. This might be something that needs to be addressed (or already addressed) in #8437 

The actual issue seems to arise from the validator - the `redirects` validator only allows the content-type to be `application/json`, but in my reproduction of the error, the browser (Chrome [canary], Edge, and Opera) sent the content-type `application/octet-stream` which is causing the issue. This error did _not_ occur in FireFox

tl;dr: The [overrides](https://github.com/TryGhost/Ghost/blob/master/core/server/config/overrides.json#L63) file needs to add `application/octet-stream` as an allowed `content-type`, or the XML Request needs to set the type manually.

**Comment by kirrg001** [comment]
@MertcanGokgoz Thanks for your report, can confirm that is a windows bug.

@vikaspotluri123 Thanks for the details. 

A PR is very welcome 🙃
